api/commands/compile.ts

```diff
@@ -97,10 +97,14 @@ export class CompileChildProcess {
       const pipe = this.childProcess.stdio[4] as Readable;
       const chunks: Buffer[] = [];
       pipe?.on("readable", () => {
-        let buffer: Buffer = pipe.read();
-        while (buffer) {
-          chunks.push(buffer);
+        let firstRead = true;
+        let buffer: Buffer;
+        while (buffer || firstRead) {
+          firstRead = false;
           buffer = pipe.read();
+          if (!!buffer) {
+            chunks.push(buffer);
+          }
```

Comment:
```suggestion
        let buffer: Buffer = pipe.read();
        while (buffer) {
          chunks.push(buffer);
          buffer = pipe.read();
        }
```

Reply:
Reading when there's nothing in the pipe throws an error; the second read can read when there's only one thing to read

Reply:
i think my code is equivalent to yours